cuBERT_topic_modelling/cuBERTopic.py

```diff
@@ -0,0 +1,360 @@
+                topic_embedding = create_embeddings(
+                    cudf.Series(" ".join(words_arr))
+                ).reshape(1, -1)
+                topic_words = mmr(
+                    topic_embedding,
+                    word_embeddings,
+                    words,
+                    top_n=n,
+                    diversity=0
+                )
+                top_n_words[topic] = [
+                    (word, value)
+                    for word, value in top_n_words[topic]
+                    if word in topic_words
+                ]
+
+        top_n_words = {
+            label: values[:n] for label, values in top_n_words.items()
+        }
+
+        topic_names = {
+            key: f"{key}_" + "_".join([word[0] for word in values[:4]])
+            for key, values in top_n_words.items()
+        }
+        return top_n_words, topic_names
+
+    def extract_topic_sizes(self, df):
+        """Calculate the topic sizes
+        Arguments:
+            documents: dataframe with documents and their corresponding IDs
+            and added Topics
+
+        Returns:
+            topic_sizes: DataFrame containing topic cluster sizes
+        """
+        topic_sizes = (
+            df.groupby(["Topic"])
+            .Document.count()
+            .reset_index()
+            .rename({"Topic": "Topic", "Document": "Count"}, axis="columns")
+            .sort_values("Count", ascending=False)
+        )
+        return topic_sizes
+
+    def fit_transform(self, data):
+        """Fit the models on a collection of documents, generate topics, and return
+        the docs with topics
+        Arguments:
+            documents: A list of documents to fit on
+
+        Returns:
+            predictions: Topic predictions for each documents
+            probabilities: The probability of the assigned topic per document.
+        """
+
+        documents = cudf.DataFrame(
+            {"Document": data, "ID": range(len(data)), "Topic": None}
+        )
+
+        # Extract embeddings
+        embeddings = create_embeddings(
+            documents.Document
+        )
+
+        # Reduce dimensionality with UMAP
+        umap_embeddings = self.reduce_dimensionality(embeddings)
+        del embeddings
+
+        # Cluster UMAP embeddings with HDBSCAN
+        documents, probabilities = self.clustering_hdbscan(
+            umap_embeddings,
+            documents
+        )
+
+        del umap_embeddings
+
+        documents = self.sort_mappings_by_frequency(documents)
+
+        tf_idf, count, docs_per_topics_topics, docs_df = self.create_topics(
+            documents
+        )
+        top_n_words, name_repr = self.extract_top_n_words_per_topic(
+            tf_idf, count, docs_per_topics_topics, n=30
+        )
+
+        self.topic_sizes_df["Name"] = self.topic_sizes_df["Topic"].map(
+            name_repr
+        )
+        self.top_n_words = top_n_words
+        predictions = documents.Topic.to_arrow().to_pylist()
```

Comment:
Any reason to convert it to `pylist` ?

Reply:
I was matching the API for `BERTopic` [as mentioned here](https://github.com/MaartenGr/BERTopic/blob/master/bertopic/_bertopic.py#L312)

I can change it to back to a `cudf.Series`, I think that should be fine if we don't want a python `list`